python/lsst/ap/association/testApdb.py

```diff
@@ -0,0 +1,575 @@
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""Standalone pipelinetask to populate an APDB with simulated data.
+"""
+
+__all__ = ("TestApdbConfig",
+           "TestApdbTask",
+           )
+
+
+import numpy as np
+import pandas as pd
+
+from lsst.daf.base import DateTime
+import lsst.dax.apdb as daxApdb
+import lsst.geom
+from lsst.meas.base import DetectorVisitIdGeneratorConfig, IdGenerator
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.ap.association.loadDiaCatalogs import LoadDiaCatalogsTask
+from lsst.ap.association.diaPipe import DiaPipelineTask
+from lsst.ap.association.utils import (
+    convertTableToSdmSchema,
+    readSchemaFromApdb,
+    column_dtype,
+    make_empty_catalog,
+)
+import lsst.sphgeom
+
+
+class TestApdbConnections(
+        pipeBase.PipelineTaskConnections,
+        dimensions=("instrument", "visit", "detector")):
+    """Butler connections for TestApdbTask.
+    """
+
+
+class TestApdbConfig(pipeBase.PipelineTaskConfig,
+                     pipelineConnections=TestApdbConnections):
+    """Config for TestApdbTask.
+    """
+    apdb_config_url = pexConfig.Field(
+        dtype=str,
+        default=None,
+        optional=False,
+        doc="A config file specifying the APDB and its connection parameters, "
+            "typically written by the apdb-cli command-line utility. "
+            "The database must already be initialized.",
+    )
+    survey_area = pexConfig.Field(
+        dtype=float,
+        default=10,
+        doc="Area (in degrees) of the simulated survey",
```

Comment:
Hmm, with the area so small the distribution across spatial partitions will be very uneven, that will likely cause uneven load across Cassandra cluster.

Reply:
Yes, this was just for testing. The actual value will be 18000.